# Explore here

In [1]:
# Your code hereimport pandas as pd
import pandas as pd


total_data = pd.read_csv("https://raw.githubusercontent.com/4GeeksAcademy/NLP-project-tutorial/main/url_spam.csv")
total_data.head()

url  is_spam
0  https://briefingday.us8.list-manage.com/unsubs...     True
1                             https://www.hvper.com/     True
2                 https://briefingday.com/m/v4n3i4f3     True
3   https://briefingday.com/n/20200618/m#commentform    False
4                        https://briefingday.com/fan     True

In [2]:
#convertimos la columna numerica a entrea true = 1 false = 0
total_data['is_spam'] = total_data['is_spam'].astype(int)
total_data.head()

url  is_spam
0  https://briefingday.us8.list-manage.com/unsubs...        1
1                             https://www.hvper.com/        1
2                 https://briefingday.com/m/v4n3i4f3        1
3   https://briefingday.com/n/20200618/m#commentform        0
4                        https://briefingday.com/fan        1

In [3]:
total_data['is_spam'].value_counts()

is_spam
0    2303
1     696
Name: count, dtype: int64

In [4]:
print(f"Nulos:{total_data.isna().sum()}")
print(f"duplicados : {total_data.duplicated().sum()}")


Nulos:url        0
is_spam    0
dtype: int64
duplicados : 630


In [5]:
total_data = total_data.drop_duplicates()

In [6]:
total_data['is_spam'].value_counts()


is_spam
0    2125
1     244
Name: count, dtype: int64

Podemos ver que los datos no estan balanceados tenemos 2125 datos de control y apenas 244 que son spam

me di el trabajo de buscar otro dataset que contenga link  clasificados de igual manera para agregar link qeu sean spam y  balancear los datos 

In [8]:
spam_df = pd.read_csv(r'C:\Users\jdean\OneDrive\Escritorio\pythonmysql\NPL_JOEL\url_spam_classification.csv')

In [10]:
spam_df['is_spam'] = spam_df['is_spam'].astype(int)

In [15]:
spam_df = spam_df.drop_duplicates()


In [16]:
spam_df['is_spam'].value_counts()

is_spam
0    84796
1     2785
Name: count, dtype: int64

In [18]:
spam_df = spam_df[spam_df['is_spam'] == 1]


In [23]:
spam_df = spam_df.sample(1900)

In [24]:
total_data = pd.concat([total_data, spam_df], ignore_index=True)

In [25]:
total_data['is_spam'].value_counts()


is_spam
1    2144
0    2125
Name: count, dtype: int64

In [26]:
print(f"Nulos:{total_data.isna().sum()}")
print(f"duplicados : {total_data.duplicated().sum()}")

Nulos:url        0
is_spam    0
dtype: int64
duplicados : 168


In [28]:
total_data = total_data.drop_duplicates()
print(f"duplicados : {total_data.duplicated().sum()}")

duplicados : 0


In [29]:
total_data['is_spam'].value_counts()

is_spam
0    2125
1    1976
Name: count, dtype: int64

Ahora tenemos un dataframe balanceado 

In [30]:
total_data.to_csv(r'C:\Users\jdean\OneDrive\Escritorio\pythonmysql\NPL_JOEL\url_spam_Ejercicio.csv', index=False)